src/core/server/ui_settings/ui_settings_client.ts

```diff
@@ -147,17 +143,50 @@ export class UiSettingsClient implements IUiSettingsClient {
     return defaultsDeep(userProvided, this.defaults);
   }
 
+  private validateKey(key: string, value: unknown) {
+    const definition = this.defaults[key];
+    if (value === null || definition === undefined) return;
```

Comment:
I think we should throw when set a key without definition. Should I add this logic now or add to the list of future improvements?

Reply:
I agree that would be the preferred behaviour, but it feels like a risky change to make, it could easily cause exceptions in code. Maybe we should start with logging a message similar to setting an invalid key.

Reply:
I think there's at least two cases where introducing an exception would be a problem:
- Any settings that have been removed from the codebase but are still present in the user's config
- Any settings that belong to a plugin that was enabled but is now disabled. I don't think we should ever break Kibana in this case.